src/sas/sasgui/perspectives/calculator/model_editor.py

```diff
@@ -1112,14 +1115,20 @@ def get_param_helper(self, line):
         """
         items = line.split(";")
         for item in items:
-            name = item.split("=")[0].lstrip().rstrip()
+            name = item.split("=")[0].strip()
+            description = ""
             try:
-                value = item.split("=")[1].lstrip().rstrip()
+                value = item.split("=")[1].strip()
+                if value.count("#"):
+                    # If line ends in a comment, remove it before parsing float
+                    index = value.index("#")
+                    description = value[(index + 1):].strip()
+                    value = value[:value.index("#")].strip()
                 float(value)
-            except:
+            except ValueError:
                 value = 1.0 # default
```

Comment:
Maybe use a regular expression?

```
>>> import re
>>> pat = re.compile(r"^ *(?P<name>[A-Za-z]\w*) *(= *(?P<value>[-0-9eE.+]+) *)?(# *(?P<desc>.*) *)?$")
>>> M = pat.match(" a "); (M.group('name'), M.group('value'), M.group('desc'))
('a', None, None)
>>> M = pat.match(" a # hello "); (M.group('name'), M.group('value'), M.group('desc'))
('a', None, 'hello ')
>>> M = pat.match(" a = -1.e-4# hello "); (M.group('name'), M.group('value'), M.group('desc'))
('a', '-1.e-4', 'hello ')
>>> M = pat.match(" a = -1.e-4 "); (M.group('name'), M.group('value'), M.group('desc'))
('a', '-1.e-4', None)
```

Recognizing all and only floating point and integer for the value makes for a very complicated regular expression, so you may still fail on float(value).

If the line doesn't match or if it matches and the value conversion fails, shouldn't the user be told?  Currently it will silently put in a value of 1.